### DISCUSSION TITLE
Easier Way To Download!?

### DISCUSSION THREAD

**Discussion by justinelliotmeyers** [OvertureMaps/data#3] [Discussion]
Is there an easier way to download these? I understand the data is massive but the few ways that are setup are going to be very difficult for the masses to access. 

Appreciate any other options your teams can put out for the average user

**Comment by lharless** [comment]
The way I did it is this.

*note* I have no AWS account.

Download the Amazon AWS command line tool, then run `aws s3 cp --region us-west-2 --no-sign-request --recursive s3://overturemaps-us-west-2/release/2023-07-26-alpha.0/ <DESTINATION>`
where destination is your local directory, like d:\overturemaps

It will download the datasets with no further prompting.

The entire dataset is 215.5 GB as of this writing.

**Reply by selimsezr** [reply]
What is the type of downloaded file? How can I see it on the map in the easiest way?

**Reply by mtravis** [reply]
The data is currently only available as parquet. 

You can follow this blog on how to convert the parquet data to a format like geojson or geopackage using DuckDB.

https://til.simonwillison.net/overture-maps/overture-maps-parquet

**Reply by druskacik** [reply]
If you are just downloading the `Places` part, the total size pretty small (about 6GB). The command to download:

```
aws s3 cp --region us-west-2 --no-sign-request --recursive s3://overturemaps-us-west-2/release/2024-03-12-alpha.0/theme=places/ data/places
```

**Comment by jiayuasu** [comment]
@justinelliotmeyers

This can be easily done by the GeoParquet version of OvertureMap data produced by Apache Sedona. Please refer to my blog post: https://medium.com/@dr.jiayu/harnessing-overture-maps-data-apache-sedonas-journey-from-parquet-to-geoparquet-d99f7767a499

## Wherobots uses Sedona to generate these data and provide them for free. 

Buildings: `s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=buildings/type=building`
Places: `s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=places/type=place`
AdminBoundary: `s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=admins/type=administrativeBoundary`
AdminLocality: `s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=admins/type=locality`
Transportation Connector: `s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=transportation/type=connector`
Transportation Segment: `s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=transportation/type=segment`

## Method 1: use Sedona to download data according to the exact shape of a region

Sedona supports geospatial filter pushdown on geoparquet so you can easily download the data of a country using its real shape.

Example spatial SQL queries:

```
spatial_filter = "POLYGON ((-122.235128 47.650163, -122.233796 47.65162, -122.231581 47.653287, -122.228514 47.65482, -122.227526 47.655204, -122.226175 47.655729, -122.222039 47.656743999999996, -122.218428 47.657464, -122.217026 47.657506, -122.21437399999999 47.657588, -122.212091 47.657464, -122.212135 47.657320999999996, -122.21092999999999 47.653552, -122.209834 47.650121, -122.209559 47.648976, -122.209642 47.648886, -122.21042 47.648658999999995, -122.210897 47.64864, -122.211005 47.648373, -122.21103099999999 47.648320999999996, -122.211992 47.64644, -122.212457 47.646426, -122.212469 47.646392, -122.212469 47.646088999999996, -122.212471 47.645213, -122.213115 47.645212, -122.213123 47.644576, -122.21352999999999 47.644576, -122.213768 47.644560999999996, -122.21382 47.644560999999996, -122.21382 47.644456999999996, -122.21373299999999 47.644455, -122.213748 47.643102999999996, -122.213751 47.642790999999995, -122.213753 47.642716, -122.213702 47.642697999999996, -122.213679 47.642689999999995, -122.21364 47.642678, -122.213198 47.642541, -122.213065 47.642500000000005, -122.212918 47.642466, -122.21275 47.642441, -122.212656 47.642433, -122.21253899999999 47.642429, -122.212394 47.64243, -122.212182 47.642444999999995, -122.211957 47.642488, -122.211724 47.642551999999995, -122.21143599999999 47.642647, -122.210906 47.642834, -122.210216 47.643099, -122.209858 47.643215, -122.20973000000001 47.643248, -122.20973599999999 47.643105, -122.209267 47.643217, -122.208832 47.643302, -122.208391 47.643347999999996, -122.207797 47.643414, -122.207476 47.643418, -122.20701199999999 47.643397, -122.206795 47.643387999999995, -122.205742 47.643246, -122.20549 47.643201999999995, -122.20500200000001 47.643119, -122.204802 47.643085, -122.204641 47.643066, -122.204145 47.643012, -122.203547 47.643012, -122.203097 47.643107, -122.20275699999999 47.643283, -122.202507 47.643496999999996, -122.202399 47.643653, -122.202111 47.643771, -122.201668 47.643767, -122.201363 47.643665, -122.20133 47.643648999999996, -122.201096 47.643536, -122.200744 47.64328, -122.200568 47.64309, -122.200391 47.642849, -122.200162 47.642539, -122.199896 47.642500000000005, -122.19980799999999 47.642424, -122.199755 47.642376999999996, -122.199558 47.642227999999996, -122.199439 47.642157, -122.199293 47.642078999999995, -122.199131 47.642004, -122.198928 47.641925, -122.19883 47.641892, -122.19856300000001 47.641811999999994, -122.198203 47.641731, -122.197662 47.641619999999996, -122.196819 47.641436, -122.196294 47.641309, -122.196294 47.642314, -122.19628 47.642855, -122.196282 47.642897999999995, -122.196281 47.643111, -122.196283 47.643415, -122.196283 47.643508999999995, -122.19628399999999 47.643739, -122.196287 47.644203999999995, -122.196287 47.644262999999995, -122.19629 47.644937999999996, -122.19629 47.644954999999996, -122.196292 47.645271, -122.196291 47.645426, -122.19629499999999 47.646315, -122.19629499999999 47.646432, -122.195925 47.646432, -122.195251 47.646432, -122.190853 47.646429999999995, -122.187649 47.646428, -122.187164 47.646426, -122.18683 47.646426, -122.185547 47.646409, -122.185546 47.646316, -122.185537 47.645599, -122.185544 47.644197, -122.185537 47.643294999999995, -122.185544 47.642733, -122.185541 47.641757, -122.185555 47.640681, -122.185561 47.63972, -122.185557 47.638228999999995, -122.185591 47.635419, -122.185611 47.634750999999994, -122.18562299999999 47.634484, -122.18561700000001 47.634375999999996, -122.185592 47.634311, -122.185549 47.634232999999995, -122.185504 47.634181999999996, -122.185426 47.634119, -122.184371 47.633424999999995, -122.18400000000001 47.633198, -122.183896 47.633134, -122.1838 47.633067, -122.18375499999999 47.633019999999995, -122.183724 47.632959, -122.183695 47.632858, -122.183702 47.632675, -122.182757 47.632622999999995, -122.182365 47.63259, -122.18220600000001 47.632562, -122.181984 47.632504999999995, -122.18163799999999 47.632363, -122.18142 47.632262999999995, -122.181229 47.632165, -122.181612 47.632172999999995, -122.18271899999999 47.632151, -122.183138 47.632135, -122.18440000000001 47.632081, -122.184743 47.632065999999995, -122.185312 47.63205, -122.185624 47.632047, -122.185625 47.631873999999996, -122.184618 47.63187, -122.184291 47.631878, -122.184278 47.631817999999996, -122.183882 47.629942, -122.182689 47.623548, -122.182594 47.622789999999995, -122.182654 47.622155, -122.183135 47.622372999999996, -122.183471 47.622506, -122.18360200000001 47.622552, -122.183893 47.622637999999995, -122.184244 47.62272, -122.184618 47.622777, -122.184741 47.622727999999995, -122.184605 47.622679, -122.18424 47.622622, -122.183985 47.622569, -122.183717 47.622501, -122.183506 47.622439, -122.18327 47.622357, -122.18305699999999 47.622271999999995, -122.182669 47.622088999999995, -122.182796 47.621545, -122.18347 47.619628999999996, -122.18365 47.619098, -122.183859 47.6184, -122.183922 47.617793999999996, -122.183956 47.617292, -122.183792 47.616388, -122.183261 47.614391999999995, -122.183202 47.613802, -122.183209 47.613155, -122.183436 47.612384999999996, -122.18395100000001 47.610445999999996, -122.184338 47.60924, -122.184657 47.609116, -122.18481 47.609051, -122.18491900000001 47.608987, -122.184974 47.608942, -122.185047 47.608846, -122.185082 47.608743999999994, -122.185109 47.608526999999995, -122.185116 47.608359, -122.18513 47.608315999999995, -122.185157 47.608273999999994, -122.185183 47.608247, -122.185246 47.608214, -122.185354 47.608196, -122.185475 47.608191999999995, -122.185472 47.606697, -122.185472 47.606373999999995, -122.185521 47.606272, -122.185528 47.606210999999995, -122.185506 47.606037, -122.185451 47.605872999999995, -122.185411 47.605781, -122.185358 47.605681999999995, -122.185248 47.605509999999995, -122.185127 47.605365, -122.185058 47.605292, -122.184772 47.605038, -122.184428 47.604834, -122.184122 47.604693999999995, -122.183775 47.604574, -122.183644 47.604546, -122.183708 47.604400999999996, -122.183749 47.604223999999995, -122.18376 47.604037, -122.183707 47.603778, -122.183619 47.603556999999995, -122.183559 47.603406, -122.183488 47.603303, -122.183824 47.603167, -122.184108 47.603052, -122.184478 47.602902, -122.18543 47.602495, -122.186669 47.601957, -122.186433 47.601220999999995, -122.186341 47.601127999999996, -122.18874199999999 47.593742999999996, -122.188434 47.592338999999996, -122.188479 47.591786, -122.188217 47.591269999999994, -122.18795399999999 47.590871, -122.186822 47.589228, -122.187421 47.589228999999996, -122.18848299999999 47.589228999999996, -122.188433 47.587922999999996, -122.18990000000001 47.588547, -122.191368 47.589169999999996, -122.19158 47.589222, -122.191779 47.589254999999994, -122.192117 47.589289, -122.191569 47.587478999999995, -122.191323 47.586628999999995, -122.191295 47.586554, -122.191268 47.586479, -122.191192 47.586318, -122.191163 47.586268999999994, -122.1911 47.586164, -122.19099 47.586011, -122.19067 47.585668999999996, -122.1905 47.585515, -122.190301 47.58531, -122.190143 47.585152, -122.189573 47.584576999999996, -122.188702 47.583735999999995, -122.188646 47.583679, -122.188239 47.583258, -122.188037 47.583005, -122.187832 47.582657, -122.187726 47.582164999999996, -122.18769499999999 47.581964, -122.18768299999999 47.581781, -122.187678 47.581592, -122.18766099999999 47.581455, -122.187674 47.581311, -122.18768 47.581146, -122.187722 47.580877, -122.187817 47.580569999999994, -122.187932 47.580301999999996, -122.188047 47.580087, -122.188161 47.579933999999994, -122.188399 47.579660999999994, -122.18851699999999 47.579547, -122.188621 47.579454, -122.188042 47.579493, -122.18762 47.579527, -122.187806 47.579358, -122.188009 47.579175, -122.18814499999999 47.579051, -122.188177 47.579021, -122.18842000000001 47.5788, -122.188638 47.578461, -122.188895 47.57806, -122.189791 47.577281, -122.190008 47.577103, -122.190372 47.576805, -122.19119 47.576358, -122.191877 47.576087, -122.193025 47.57566, -122.194317 47.575185999999995, -122.196061 47.574664, -122.197239 47.574386999999994, -122.197873 47.574267, -122.198286 47.574189999999994, -122.199091 47.574044, -122.199067 47.574574999999996, -122.199007 47.575921, -122.200335 47.578222, -122.20057299999999 47.578345999999996, -122.2009 47.578517999999995, -122.201095 47.578621999999996, -122.20138399999999 47.578776999999995, -122.201465 47.57882, -122.201516 47.578846999999996, -122.205753 47.581112, -122.209515 47.583124, -122.210634 47.583721, -122.21473399999999 47.587021, -122.21538699999999 47.588254, -122.21580399999999 47.589042, -122.216534 47.590421, -122.220092 47.596261, -122.220434 47.596821, -122.22041899999999 47.597837999999996, -122.220289 47.606455, -122.220234 47.610121, -122.22048 47.615221999999996, -122.220359 47.615379, -122.220283 47.615477999999996, -122.21999 47.615854999999996, -122.219993 47.61597, -122.22023300000001 47.616634, -122.220356 47.616687999999996, -122.220409 47.616712, -122.221401 47.618538, -122.22142 47.618573, -122.221456 47.618635, -122.221791 47.619222, -122.222492 47.619682999999995, -122.222799 47.619886, -122.222083 47.620368, -122.222046 47.620407, -122.222028 47.620449, -122.222025 47.620483, -122.22203999999999 47.620523999999996, -122.222079 47.620557999999996, -122.222156 47.620594999999994, -122.222458 47.620629, -122.222454 47.620673, -122.222454 47.620711, -122.22244599999999 47.621041999999996, -122.223056 47.621041, -122.223129 47.62104, -122.223153 47.62104, -122.223574 47.621041, -122.22377900000001 47.621041, -122.223857 47.621041, -122.22467499999999 47.621041, -122.224712 47.62104, -122.224958 47.62104, -122.225167 47.621049, -122.226882 47.621037, -122.227565 47.621032, -122.228002 47.621029, -122.22797800000001 47.621300999999995, -122.227919 47.626574999999995, -122.227914 47.627085, -122.227901 47.6283, -122.227881 47.630069, -122.227869 47.631177, -122.227879 47.631952999999996, -122.22789 47.633879, -122.227886 47.63409, -122.227871 47.635534, -122.227918 47.635565, -122.228953 47.635624, -122.22895199999999 47.635571999999996, -122.231018 47.635574999999996, -122.233276 47.635588999999996, -122.233287 47.63617, -122.233273 47.63639, -122.233272 47.636469999999996, -122.23327 47.636578, -122.233266 47.636827, -122.233263 47.636851, -122.233262 47.637014, -122.23322999999999 47.638110999999995, -122.233239 47.638219, -122.233262 47.638279, -122.233313 47.638324999999995, -122.233255 47.638359, -122.233218 47.638380999999995, -122.233153 47.638450999999996, -122.233136 47.638552999999995, -122.233137 47.638692, -122.232715 47.639348999999996, -122.232659 47.640093, -122.232704 47.641375, -122.233821 47.645111, -122.234906 47.648874, -122.234924 47.648938, -122.235128 47.650163))"

df = sedona.read.format("geoparquet").load("s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=buildings/type=building")
df = df.filter("ST_Contains(ST_GeomFromWKT('"+spatial_filter+"'), geometry) = true")
df.count()
df.write.format("geoparquet").save("/MyPath/GeoParquet_File_Name.parquet")
```

Try it now. This query on the 120GB Building dataset will take about 3 min to finish on the docker image.

```
docker pull apache/sedona
docker run -p 8888:8888 apache/sedona:latest
```

The Jupyter notebook will be available at https://localhost:8888/

## Method 2: Use AWS CLI to download a specific region according to GeoHash ID:

All GeoParquet data provided by Wherobots are partitioned by GeoHash ID with 2 character precision.  See below:
<img width="1397" alt="image" src="https://github.com/OvertureMaps/data/assets/10948864/4e761672-318b-4473-b8ca-4b09760eba71">


You can download a specific region as follows

```
aws s3 cp --region us-west-2 --no-sign-request --recursive s3://wherobots-public-data/overturemaps-us-west-2/release/2023-07-26-alpha.0/theme=buildings/type=building/geohash=gb/ <DESTINATION>
```

You can find the geohash id of your target geospatial region using this online tool: https://www.movable-type.co.uk/scripts/geohash.html

**Reply by ghost** [reply]
`docker run -p 8888:8888 apache/sedona:latest`

this return `ssh:unrecognized service`
why is that so? Any help much appreciated

**Reply by jiayuasu** [reply]
@justinelliotmeyers I am not exactly sure why. Can you try apache/sedona:1.5.1? I tried both arm64 and amd64 version on my Mac Apple Silicon and they all work fine.

**Reply by rbavery** [reply]
@Jigeeshu can you paste the full traceback? Are you navigating to the localhost url provided by the docker image after it is running?

On WSL or Powershell, after running the container, I navigate to  `http://127.0.0.1:8888/lab` and can access the jupyter notebook running on the image.

**Reply by ghost** [reply]
@rbavery , Thank you for your reply.  I am having issues running the docker container.  When I try to run the docker container I get server error (HTTP code 500) saying the port 8080 is not available. Even if I kill the process running on 8080 it still shows already in use. It is running some java command is all I can make of it.

**Reply by rbavery** [reply]
can you give more details about your system (OS, version)? How are you running the container? In a VM, WSL, Powershell, Linux terminal?

Pasting the terminal output would also help. It could be your system has something running on port 8080 already that i conflicting with the Sedona image.

**Reply by jiayuasu** [reply]
@justinelliotmeyers When run the Sedona docker image, the only really needed port of the host is 8888. Other ports are optional.

Minimal command:
```
docker run -p 8888:8888 apache/sedona:latest
```


Full command
```
docker run -e DRIVER_MEM=6g -e EXECUTOR_MEM=8g -p 8888:8888 -p 8080:8080 -p 8081:8081 -p 4040:4040 apache/sedona:latest
```

**Comment by d3netxer** [comment]
@jiayuasu I am trying to run the docker image:

```
docker pull apache/sedona
docker run -p 8888:8888 apache/sedona:latest
```
but when I go to https://localhost:8888/ it says Token authentication is enabled but I don't see any token.

**Reply by jiayuasu** [reply]
@d3netxer This is strange. I know a few folks tested it several times and it worked fine.

Can you try (1) open it in a incognito window (2) when you start it in the terminal, can you check which address the jupyter lab is binded to, then try those addresses? In my terminal, it is as follows:

![image](https://github.com/OvertureMaps/data/assets/10948864/53acfe8b-1fa0-4245-bd9e-29d0c1b8f419)


In addition, we provide images for both amd64 and arm64. I believe this serves 90% of use cases such as Intel 64-bit chips and Apple Silicon. But please double check.

**Reply by d3netxer** [reply]
Thanks a lot for your help @jiayuasu, I figured out that I had an unrelated jupyter notebook already running on port 8888. I stopped the other notebook and it is working fine now.

**Comment by yharby** [comment]
@d3netxer , @jiayuasu I'm working on simplifying the data download process in a fork of the @wherobots repository, which you can find here: https://github.com/Youssef-Harby/OvertureMaps. Once I've completed the changes, I plan to submit a pull request to the original repository at https://github.com/wherobots/OvertureMaps. To test it out, you can use the Docker Compose file included in my fork.

After cloning the repo and cd to it
```bash
docker compose up -d 
```
http://localhost:8888/lab

after you run the part of getting the df you can save the Sedona DataFrame as a `GeoParquet` then you can convert it to GeoPackage/GeoJSON…etc. with `ogr2ogr`

example of save the df of building to offline parquet (it will go directly to your current project folder because it's mounted volume to the host directory):
```python
df_building.write.format("geoparquet").save("/opt/workspace/building.parquet")
```

to convert it you can use for example :
```bash
ogr2ogr -f GPKG output.gpkg building.parque
```

**Comment by d3netxer** [comment]
unfortunately it is not working for me @Youssef-Harby. I am getting an error when tying the Docker compose command:

```
e8126f49e92b85724caf8ee0c30b5a3e92a9a149ee91223c51f  403  Forbidden [IP: 52.218.201.234 443]
33.62 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
------
failed to solve: process "/bin/sh -c apt-get install -y -V libarrow-dev libparquet-dev" did not complete successfully: exit code: 100
```

**Reply by yharby** [reply]
try to comment the build part in the `docker-compose.yml` and use official apache/sedona docker image:
like:
```docker
version: '3.8'

services:
  sedona:
    container_name: sedona
    # build:
    #   context: .
    #   dockerfile: Dockerfile
    image: apache/sedona:latest
    ports:
      - "8888:8888"
    volumes:
      - .:/opt/workspace/
```

**Reply by d3netxer** [reply]
Thank you @Youssef-Harby, I used the official apache/sedona image and it is working. I see in your fork you have added your own Dockerfile. I can't find the Dockerfile of the official apache/sedona image so I can't compare it. What are you trying to accomplish in your fork? How are you trying to simplify the data download process?

**Reply by yharby** [reply]
@d3netxer, I am attempting to build the latest version of GDAL/OGR with the GeoParquet Driver enabled for conversion to and from GeoParquet and other extensions. I am extending the official Sedona Docker image with the following Dockerfile:

```Dockerfile
FROM apache/sedona

# Update and install build tools and Python
RUN apt-get update && \
    apt-get install -y build-essential cmake libboost-all-dev

# Install wget separately for debugging
RUN apt-get install -y wget

# Define the GDAL version here
ARG GDAL_VERSION=3.7.1

# Install Apache Arrow
RUN apt-get install -y -V ca-certificates lsb-release wget
RUN wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
RUN apt-get install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
RUN apt-get update
RUN apt-get install -y -V libarrow-dev libparquet-dev

WORKDIR /home

# Download GDAL
RUN wget http://download.osgeo.org/gdal/${GDAL_VERSION}/gdal-${GDAL_VERSION}.tar.gz

# Unzip and build GDAL with specific drivers enabled, then remove the tar file
RUN tar -xzvf gdal-${GDAL_VERSION}.tar.gz && \
    cd gdal-${GDAL_VERSION} && \
    mkdir build && \
    cd build && \
    cmake .. -DGDAL_USE_ARROW=ON -DGDAL_USE_PARQUET=ON && \
    cmake --build . && \
    cmake --build . --target install && \
    cd /home && \
    rm gdal-${GDAL_VERSION}.tar.gz

# Update environment variables
ENV PATH="/home/gdal-${GDAL_VERSION}/build:${PATH}"
ENV LD_LIBRARY_PATH="/home/gdal-${GDAL_VERSION}/build:${LD_LIBRARY_PATH}"
ENV GDAL_DATA="/home/gdal-${GDAL_VERSION}/data:${GDAL_DATA}"

# Update shared library cache
RUN ldconfig

# Install Python GDAL package
RUN pip3 install GDAL==3.7.1.1
```
The official Sedona Docker image is using GDAL version 3.4.1, which does not support Arrow or Parquet/GeoParquet.
https://github.com/apache/sedona/blob/master/docker/spark.sh#L50-L51

I am currently encountering issues related to the Parquet build. I can confirm that both Parquet and Arrow have been built successfully. When I run the command `ogrinfo --formats`, I can see that the Arrow and Parquet drivers are successfully installed. However, when I attempt to convert a `building.parquet` file to GPKG using the command `ogr2ogr -f GPKG output.gpkg building.parquet`, I encounter the following error message:

```bash
Unable to open datasource 'building.parquet' with the following drivers.
  ...
  -> `Parquet'
  -> `Arrow'
  ...
```
It could potentially be an issue related to Docker file permissions or the GDAL/OGR build process; however, I am not certain about the exact cause at this point.

**Reply by mtravis** [reply]
@Youssef-Harby I've been working moving to parquet to postgis over the last few weeks using ogr2ogr and the notes here.
https://gdal.org/drivers/vector/parquet.html

The fact you can see parquet in the list of formats makes me think it's a permissions issue. Have you tried running with sudo if possible. Are you in the  correct directory to access buildings.parquet? 

Another thing to try is running with better logging like so
`ogr2ogr -f GPKG output.gpkg building.parquet --debug ON`

**Reply by yharby** [reply]
@mtravis I did consider permissions issues and tried running the command with elevated permissions (i.e., sudo), but the issue persisted. I also double-checked the directory paths to make sure the building.parquet file is accessible. I'll try running the command with debug logging enabled, as you suggested, to get more information on what might be causing the problem.
note: it's working on my local machine but without  :) , I am trying to create a docker to make it easier in pipeline and automation process also to help more people to easily download and convert new Parquet and OvertureMaps Data.

since you mentioned you've been working on a similar task, it would be great to collaborate

**Reply by jiayuasu** [reply]
@Youssef-Harby Would you please push your update to the official Sedona dockerfile, if you eventually figure out the solution? Our Sedona community will really appreciate your contribution. The official Sedona docker image will also bring more visibility to your contribution.

**Reply by yharby** [reply]
@jiayuasu Once I've resolved the issue, I'll be more than happy to push the update to the official Sedona Dockerfile. I believe that having a robust and reliable Docker image will indeed benefit the community and streamline the pipeline and automation processes.

I'm also excited about the possibility of collaborating on similar tasks. Your expertise and the collective knowledge of the Sedona community would undoubtedly make the process more efficient and the solution more robust.

I'll keep you updated on my progress and will reach out for a review once the update is ready for the official Docker image.
